### PR TITLE
Support scala 2.12.0-M3

### PR DESCRIPTION
**Pull request by 2m**
Also use sbt-git for deriving versions.

### CODE REVIEW

**Comment by ktoso** [comment]
LGTM, AFAICS

File: .travis.yml [hunk 1]
```diff
@@ -3,5 +3,6 @@ sudo: false
 language: scala
 jdk:
   - openjdk6
+  - oraclejdk8
```
Comment (rkuhn) [inline .travis.yml]: what exactly does this do? and who is Travis? ;-)

File: plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala [hunk 2]
```diff
@@ -122,7 +122,7 @@ trait AST { this: TransformCake ⇒
         case m: MethodSymbol => m.isVarargsMethod
         case _               => false
       }
-      val d = newDefDef(sym, EmptyTree)()
+      val d = DefDef(sym, EmptyTree)
```
Comment (rkuhn) [inline plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala]: does this work with all cross-versions?

File: .travis.yml (hunk 1 above)
Comment (2m) [inline .travis.yml]: This runs tests on Java 6 and Java 8. In the build file it checks the current version of Java and adds scala 2.12.0-MX versions to `crossScalaVersions` only if it detects that it is running on Java 8.

File: plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala (hunk 2 above)
Comment (2m) [inline plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala]: Yes, it passes all the tests with all cross scala versions. Unfortunately Mr. Travis build got a timeout while switching to the latest cross scala version.

Comment (ktoso) [inline plugin/src/main/scala/com/typesafe/genjavadoc/AST.scala]: Oh lemme try that then.

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=0.13.1
+sbt.version=0.13.9
```
Comment (rkuhn) [inline project/build.properties]: 11 :-)

**Comment by dotta** [comment]
Subsumed by #67

**Comment by 2m** [comment]
Retook this one as it needed some more work. All tests passed now. Yay!

**Pull request merged**